File: Algorithm/BinarySearch.java
import java.util.* ;
import java.util.Scanner;

public static Boolean binarySearch(int[] arr, int n){

int start=0 ,end=arr.length-1,mid=arr.length/2;

if(arr.length==0){
    return false;
}
for(int i=0;i<arr.length;i++){
    mid=(start+end)/2;
    if(n>arr[mid]) start=mid+1;
    
    else if(n<arr[mid]) end=mid-1;
    
    else return true;
}
return false;
}

File: Algorithm/BinarySearch.cpp
#include<bits/stdc++.h>
using namespace std;
int BinarySearch(int arr[],int n,int x)
{int high=n-1;
  int low=0;
  int mid=(high+low)/2;
  while(low<=high)
  { 
  if(x==arr[mid])
    return mid;
  else if(x<arr[mid])
     high=mid-1;
   else if(x>arr[mid])
     low=mid+1;}
     return -1;}
  int main()
  {int n;
   cin>>n;
   int arr[n];  /*This array must be sorted for binary search to work*/
   int i;
   for(i=0;i<n;i++)
    cin>>arr[i];
    cin>>x;
    cout<<BinarySearch(arr,n,x);}
  
